### ISSUE TITLE
Carousel improvements

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#9481]
The Carousel component provides a generic UI for various carousel, such as product carousels or images. The carousel is reusable and allows for bespoke templates per carousel type. There are a number of new ux and performance improvements  we like to introduce to the carousel: 
- [x] Lazy loaded carousel items; currently all items are always loaded, which has a very bad impact on performance;
- [x] Prefetch the carousel items of the next slide when the user starts interacting (hover, focus) with the current slide or indicators
- [x] Apply ghost designs, to provide an initial skeleton of the UI
- [x] Improve keyboarding; we like the user to easily navigate through the items. 
- [x] Improve responsive styling. Instead of using fixed width per item, we rather use a pure flexbox solution, where the layout differs per breakpoint / layout. The current setup is already flexible, but changing the style requires a component change, we like to make it purely based on css.
- [x] Introduce touch for mobile users, specifically the ability to swipe through items. 
- [x] Introduce the use of `scroll-snap-type` to improve the experience.

### Accessibility
- carousel next/previous buttons are not accessible
- focusable carousel item do not come in the view when they're outside the current slide; hidden items are focusable

## Performance
- all carousel items are loaded and rendered up front

## UI/UX issues
- we only have item on xs screen
- next/previous buttons seem in active all the time
- no touch navigation on mobile available (swipe left/right)
- next slides can have only 1 item
